__tests__/users.ts

```diff
@@ -2526,3 +2528,15 @@ describe('mutation acceptFeatureInvite', () => {
     expect(invite.count).toEqual(1);
   });
 });
+
+describe('user_create_alerts_trigger after insert trigger', () => {
+  it('should insert default alerts', async () => {
+    await con.getRepository(User).delete({});
+    const repo = con.getRepository(Alerts);
+    await repo.delete({});
+    const [user] = usersFixture;
+    await saveFixtures(con, User, [user]);
+    const alerts = await repo.findOneBy({ userId: user.id });
+    expect(alerts).toBeTruthy();
```

Comment:
You could even say equal ALERT_DEFAULT, or abstract them inline to validate defaults don't change? (non blocking)

Reply:
I checked our ALERT_DEFAULT, the `lastChangelog` has some default value that is not the same with DB defaults. Not sure why though.

Reply:
So we need to adjust the table default. Chris is right, we better fix it

Reply:
Cool let's make them match, currently users without alerts would get this ALERT_DEFAULTS returned, so we need to make them match indeed

Reply:
Yeah, I agree with that, not sure though which is correct. The new date or the specific value.

![Screenshot 2023-11-09 at 7 44 15 PM](https://github.com/dailydotdev/daily-api/assets/13744167/e4c74f30-2a39-4b21-a2c9-5b29b165a4ab)

Reply:
@sshanzel vaguely recall if it wasn't in the past it would cause side re-renders.
But maybe try it locally to see if that happens. *(maybe since now it get's created on user creation it doesn't matter anymore?)

Reply:
Don't see anything weird on registration. I will update the JS defaults to new Date. Like you mentioned, it might not matter now, also that we require registration for users.

Reply:
I am still thinking about the previous implementation, maybe we should store the value in the cache for anonymous users (viewing post page) to make it work similarly to how it was before and for existing users. Though not sure if the numbers are significant and if it would be worth it.

Reply:
From the previous implementation, since we have a static value from the ALERTS_DEFAULT, the value never changes, unlike now, when an anonymous user visits a post page, every visit would then make the `lastBanner` have a new value unlike before. After further digging, the value from alerts.lastBanner is being compared to some value in the cache, which it checks whether the last seen banner (different from `alerts.lastBanner`) is less than to your alerts value, by updating it every time, for anonymous users, the relevant property might return false all the time.